utilities/titandb/titan_db_test.cc

```diff
@@ -392,17 +396,26 @@ TEST_F(TitanDBTest, BlobFileIOError) {
       mock_env->SetFilesystemActive(true);
     }
   }
+  SyncPoint::GetInstance()->DisableProcessing();
+  mock_env->SetFilesystemActive(true);
+
   std::unique_ptr<Iterator> iter(db_->NewIterator(ReadOptions()));
+  SyncPoint::GetInstance()->EnableProcessing();
   iter->SeekToFirst();
+  ASSERT_TRUE(iter->status().IsIOError());
+  SyncPoint::GetInstance()->DisableProcessing();
+  mock_env->SetFilesystemActive(true);
+
+  iter.reset(db_->NewIterator(ReadOptions()));
+  iter->SeekToFirst();
+  ASSERT_TRUE(iter->Valid());
```

Comment:
ok, our new test caught a bug here, which is good. The iterator should invalidate immediately. It should not retry until seeing valid value. Do you mind fix it in this PR? (i.e. replace the while-loop with just `GetBlobValue()` here: https://github.com/pingcap/rocksdb/blob/tikv-3.0/utilities/titandb/db_iter.h#L34). We also need a test to test all of `SeekToFirst`, `SeekToLast`, `Seek` and `SeedForPrev`.

Reply:
test is solid. Will adding IterTest and fixing Iter in new PRs.